device/common/include/traccc/device/concepts/barrier.hpp

```diff
@@ -18,7 +18,7 @@ namespace traccc::device::concepts {
  * @tparam T The barrier-like type.
  */
 template <typename T>
-concept barrier = requires(T& b) {
+concept barrier = requires(const T& b) {
```

Comment:
I don't love this; requiring the concept to work for a non-const `T` is a stronger requirement than having it work with a const `T`. Let's not make our concepts weaker for no reason.

Reply:
Okay, I don't feel strongly about this.